frameworks/moveos-stdlib/src/natives/moveos_stdlib/measurement.rs

```diff
@@ -0,0 +1,81 @@
+use crate::natives::helpers::{make_module_natives, make_native};
+use log::info;
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::gas_algebra::InternalGas;
+use move_vm_runtime::native_functions::{NativeContext, NativeFunction};
+use move_vm_types::loaded_data::runtime_types::Type;
+use move_vm_types::natives::function::NativeResult;
+use move_vm_types::pop_arg;
+use move_vm_types::values::Value;
+use smallvec::smallvec;
+use std::collections::VecDeque;
+use std::time::UNIX_EPOCH;
+
+#[derive(Debug, Clone)]
+pub struct MeasurementGas {
+    pub base: InternalGas,
+}
+
+impl MeasurementGas {
+    pub fn zero() -> Self {
+        Self { base: 0.into() }
+    }
+}
+
+fn inject_parameter(
+    _gas_params: &MeasurementGas,
+    context: &mut NativeContext,
```

Comment:
The NativeContext is initialized each time a native function is called, a mechanism is needed to maintain gas statistics throughout the entire transaction execution process.

Reply:
Custom a Native context like `NativeEventContext` or `ObjectRuntimeContext`

Reply:
> Custom a Native context like `NativeEventContext` or `ObjectRuntimeContext`

OK, got it.